pkg/resizer/csi_resizer.go

```diff
@@ -131,14 +139,30 @@ func (r *csiResizer) CanSupport(pv *v1.PersistentVolume) bool {
 	return true
 }
 
+// Resize resizes the persistence volume given request size
+// It supports both CSI volume and migrated in-tree volume
 func (r *csiResizer) Resize(pv *v1.PersistentVolume, requestSize resource.Quantity) (resource.Quantity, bool, error) {
 	oldSize := pv.Spec.Capacity[v1.ResourceStorage]
 
-	source := pv.Spec.CSI
-	if source == nil {
-		return oldSize, false, errors.New("not a CSI volume")
+	var volumeID string
+	if pv.Spec.CSI != nil {
+		// handle CSI volume
+		source := pv.Spec.CSI
+		volumeID = source.VolumeHandle
+	} else {
+		if csitranslationlib.IsMigratedCSIDriverByName(r.name) /* && migration enabled*/ {
+			// handle migrated in-tree volume
+			csiPV, err := csitranslationlib.TranslateInTreePVToCSI(pv)
+			if err != nil {
+				return oldSize, false, fmt.Errorf("failed to translate persistenc volume: %v", err)
+			}
+			volumeID = csiPV.Spec.CSI.VolumeHandle
+		} else {
+			// non-migrated in-tree volume
+			return oldSize, false, errors.New("un-supported volume type")
```

Comment:
would be nice to have more details in this error, like what type was unsupported (or at least that it is a non-migrated in-tree volume type)